lib/api/include/irods/switch_user.h

```diff
@@ -19,35 +21,33 @@ extern "C" {
 /// - All API operations invoked via the \p RcComm will be carried out as the user that was just switched to.
 /// - Server-to-server connections will be updated or disconnected based on the remote server's version.
 ///
-/// On failure, the RcComm will remain as if this API was not invoked. If interacting with the server appears unstable,
-/// the client should disconnect and establish a new connection. To avoid unknown issues, clients are encouraged to
-/// use a new connection following an error. This provides the highest protection of data.
-///
-/// \param[in] _comm     A pointer to a RcComm.
-/// \param[in] _username \parblock
-/// The name of the iRODS user to become.
-///
-/// The passed string must meet the following requirements:
-/// - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
-/// - It must be null-terminated
-/// - It must be non-empty
-/// - It must not include the zone.
-/// \endparblock
-/// \param[in] _zone     \parblock
-/// The zone of the iRODS user to become.
-///
-/// The passed string must meet the following requirements:
-/// - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
-/// - It must be null-terminated
-/// - It must be non-empty
+/// On failure, the RcComm should be closed to avoid potential issues.
+///
+/// \param[in] _comm  A pointer to a RcComm.
+/// \param[in] _input \parblock A pointer to a SwitchUserInput.
+///
+/// SwitchUserInput member variables:
+/// - \p username: The part of the iRODS username preceding the pound sign. The following requirements must be satified:
+///     - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
+///     - It must be null-terminated
+///     - It must be non-empty
+///     - It must not include the zone
+/// - \p zone: The part of the iRODS username following the pound sign. The following requirements must be satisfied:
+///     - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
+///     - It must be null-terminated
+///     - It must be non-empty
+/// - \p update_proxy_user: If set to 1, the proxy user is updated as well. This option will result in losing the
+///                         ability to invoke rc_switch_user if this option is set to 1 and the user to switch to is
+///                         not a \p rodsadmin. This option does not affect server-to-server connections.
```

Comment:
'does not affect' -> 'will be ignored by' or 'ignored on' or 'ignored for' ?

Reply:
Updated, but went with a different phrase entirely. Willing to change to one of your suggestions if you feel strongly about it or the new words aren't clear enough.

See switch_user_types.h for new words.

Reply:
No worries. Thanks.